src/OrchardCore/OrchardCore.Abstractions/BackgroundJobs/HttpBackgroundJob.cs

```diff
@@ -36,21 +44,15 @@ public static Task ExecuteAfterEndOfRequestAsync(string jobName, Func<ShellScope
             return Task.CompletedTask;
         }
 
-        // Record the current logged in user.
-        var userPrincipal = httpContextAccessor.HttpContext.User.Clone();
-
         // Increment the active jobs count to track the number of background jobs running.
         Interlocked.Increment(ref _activeJobsCount);
 
+        // Record the current logged in user.
+        var userPrincipal = httpContextAccessor.HttpContext.User.Clone();
+
         // Fire and forget in an isolated child scope.
         _ = ShellScope.UsingChildScopeAsync(async scope =>
         {
-            scope.RegisterBeforeDispose(scope =>
-            {
-                // Decrement the active jobs count when the job is disposed.
-                Interlocked.Decrement(ref _activeJobsCount);
-            }, true);
```

Comment:
Moving this to individual calls at each return point isn't necessary. Keep in mind that this operates within a separate scope, which gets disposed after the inner scope that performs the actual work. It's safe to decrement the count during the disposal of the outer scope.